src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -151,6 +183,101 @@ private ImportFileView generateApiImport(GapicMethodContext context) {
+
+    // Get disambiguated imports.
+    for (Map.Entry<String, TypeAlias> entry : typeTable.getImports().entrySet()) {
+      imports.add(generateAppImport(entry.getKey(), entry.getValue().getNickname()));
+    }
+    return ImmutableList.<ImportFileView>builder().addAll(imports).build();
+  }
+
+  private ModelTypeTable emptyTypeTable(GapicProductConfig productConfig) {
+    return new ModelTypeTable(
+        new PythonTypeTable(productConfig.getPackageName()),
+        new PythonModelTypeNameConverter(productConfig.getPackageName()));
+  }
+
+  private List<ImportFileView> generateTypesExternalImports() {
+    return ImmutableList.of(createImport("google.gax.utils.messages", "get_messages"));
+  }
+
+  private List<ImportFileView> generateTypesStandardImports() {
+    return ImmutableList.of(createImport("__future__", "absoulute_import"), createImport("sys"));
+  }
+
+  private void populateTypeTable(Set<ProtoFile> protoFileDependencies, ModelTypeTable typeTable) {
+    for (ProtoFile protoFile : protoFileDependencies) {
+      // For python, adding a single message from the proto file to the type table will populate
+      // the type table with the correct imports.
+      ImmutableList<MessageType> messages = protoFile.getMessages();
+      if (!messages.isEmpty()) {
+        typeTable.getAndSaveNicknameFor(TypeRef.of(messages.get(0)));
+      }
+    }
+  }
+
+  private Set<ProtoFile> getProtofileDependencies(Model model) {
+    // Set up BFS.
+    Iterable<Interface> apiInterfaces = new InterfaceView().getElementIterable(model);
+    Queue<ProtoFile> protoFileQueue = new LinkedList<>();
+    for (Interface apiInterface : apiInterfaces) {
+      protoFileQueue.add(apiInterface.getFile());
+    }
+
+    // BFS to get all proto dependencies.
+    Set<ProtoFile> seenFiles = new HashSet<>();
+    while (!protoFileQueue.isEmpty()) {
+      ProtoFile current = protoFileQueue.remove();
+      seenFiles.add(current);
+      for (ProtoFile protoFile : current.getDependencies()) {
+        if (!seenFiles.contains(protoFile)) {
+          protoFileQueue.add(protoFile);
+        }
+      }
+    }
+    return seenFiles;
+  }
+
+  private List<ImportFileView> generateVersionedInitStandardImports() {
+    return ImmutableList.of(createImport("__future__", "absoulte_import"));
+  }
+
+  private List<ImportFileView> generateVersionedInitAppImports(
+      Model model,
+      GapicProductConfig productConfig,
+      PackageMetadataConfig packageConfig,
+      SurfaceNamer namer) {
+
+    Iterable<Interface> apiInterfaces = new InterfaceView().getElementIterable(model);
+    ModelTypeTable typeTable = emptyTypeTable(productConfig);
+    for (Interface apiInterface : apiInterfaces) {
+      namer.getAndSaveNicknameForGrpcClientTypeName(typeTable, apiInterface);
+    }
+
+    Set<ImportFileView> imports = new TreeSet<>(importFileViewComparator());
+    // TODO: (landrito) Make this only generate the enums module if there are enums to export.
```

Comment:
[Easy check](https://github.com/googleapis/toolkit/blob/f0762cd69fa4286fca3cfe7445b369c887ba5061/src/main/java/com/google/api/codegen/py/PythonImportHandler.java#L83). It seems like this was cause an `ImportError` if you actually try to run a GAPIC with no enums, so it seems like it would be best just to address this now.